File: resources/sample.cpp
// C++ Sample for Something Theme
#include <iostream>
#include <vector>
#include <string>

#define VERSION "1.0.0"

namespace Something {

template<typename T>
class Sample {
private:
    T _data;

public:
    Sample(T data) : _data(data) {}

    void print() {
        std::cout << "Data: " << _data << std::endl;
    }
};

} // namespace Something

int main() {
    Something::Sample<std::string> s("Hello C++");
    s.print();
    std::cout << "Version: " << VERSION << std::endl;
    return 0;
}

File: resources/sample.rs
// Rust Sample for Something Theme

/// A simple struct to hold some data.
#[derive(Debug)]
struct Sample<'a> {
    id: u32,
    name: &'a str,
}

enum Message {
    Quit,
    Write(String),
}

macro_rules! say_hello {
    () => {
        println!("Hello from a macro!");
    };
}

fn main() {
    let sample = Sample { id: 1, name: "Something" };
    println!("Sample struct: {:?}", sample);

    let msg = Message::Write(String::from("Test message"));

    say_hello!();
}